feat Click to roll against Reputation

Moonpile opened the pull request:
- Reputation value is now "text-right"
- "reputation.svg" came from "beams-aura.svg"

Still needs:
- localization
- Add Reputation to icons dialog?

Moonpile commented:
How do we want to make it obvious that Reputation is rollable?  Right now I just have the word "ROLL" there, but I figure it ought to look more like skills, but I couldn't figure out how to get the hover over shadow effect.

![image](https://user-images.githubusercontent.com/3959760/147948046-cf72a284-34b0-4905-aca4-8706e482cb38.png)

Also I've got the "other modifiers" localization from the RuneMagicCard (which I currently don't have merged).  I suppose this could go to some "common" localization group.

![image](https://user-images.githubusercontent.com/3959760/147948287-ccc7caf2-40ea-4dda-bbfb-fcc1c723cfab.png)

And I made that Reputation icon a setting like the item icons even though Reputation isn't an item.  I see no reason not to add it to that dialog, do you?  And maybe putting it on the background tab might help make it clearer that Reputation is rollable?

wake42 commented:
I agree reputation can be handled as a "fake" item. It will be a bit strange to give it an `ITEM.TypeReputation` translation key
![image](https://user-images.githubusercontent.com/4853677/147966362-0f425964-5a25-44fc-9615-efe6c1ff78a3.png)
But maybe we can live with that.

To get the hover effect you just have to add the roll here https://github.com/sun-dragon-cult/fvtt-system-rqg/blob/160aae27f89fd51e1a56a572f72dc206cda8a03b/src/rqg.scss#L569 
I see that the runeMagic roll should be added as well. And I also see that I've forgotten to remove `[data-melee-roll]`, maybe you could sneak that change in while you edit it. It was for the old meleeWeapon items.
I'm uncertain if it would be enough affordance to be able to understand that it is clickable, but maybe together with that icon it could be enough, even without "ROLL". On the player side the whole text `[*] Reputation 20%` including the icon should be clickable I think 
![image](https://user-images.githubusercontent.com/4853677/147965881-687c9f97-6e9d-480c-a72c-c8b2d890b920.png)
On the GM side only `[*] Reputation` should be clickable (to make it possible to edit the value without automatically rolling :-)

I have been toying with the idea to put reputation on the top part of the character sheet. It is a defining number of the actor, but it's probably too crowded there already...

Moonpile commented:
Fixed the hover over covering the Reputation input and added Reputation to ITEM localization.

I think we agreed that the localization can await the en.json re-org.  It'll probably become clear to me where it makes sense for that to go.

![image](https://user-images.githubusercontent.com/3959760/148144214-391ecc06-d893-49bf-a489-2968212a83c6.png)

wake42 commented on src/actors/rqgActorSheet.ts:
```diff
@@ -694,6 +695,33 @@ export class RqgActorSheet extends ActorSheet<
+          );
+          clickCount = 0;
+        } else if (clickCount === 1) {
+          await ReputationCard.show(
+            this.actor,
+            // @ts-ignore wait for foundry-vtt-types issue #1165 #1166
+            this.token
+          );
```
There needs to be a setTimeout around this, otherwise it will always show a chatcard as well. Have a look at the other listeners.

Moonpile replied:
Done

wake42 commented on src/actors/rqgActorSheet.ts:
```diff
@@ -694,6 +695,33 @@ export class RqgActorSheet extends ActorSheet<
+          const speakerName = this.token?.name || this.actor.data.token.name;
+          await CharacteristicCard.roll(
+            "Reputation",
+            this.actor.data.data.background.reputation || 0,
```
For some reason I always get 0% chance when doubleclicking to do a direct roll.

Moonpile replied:
I wasn't getting quite that behavior, but I gave ReputationCard a "directroll" method which sets up flags as if you just clicked Roll on the card without making any changes.  This works the same way as the RuneMagicCard.

wake42 reviewed (approved):
Approving, but with a comment (that can be addressed in #105).
The translation key for the reputation card modifier is `RQG.RuneMagicCard.OtherModifiers`. Is this a case where the key should be `RQG.Game.OtherModifiers` instead since it's used in more places than one?
